### PR TITLE
Enable optimization level setting

### PR DESCRIPTION
**Pull request by akharche**
Issue #39

### CODE REVIEW

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]
Please, merge with master.

**Comment by akharche** [comment]
This PR breaks GDB support when -O0 optimization level is set

**Comment by akharche** [comment]
@reazulhoque @diptorupd  If we take `llvm-spirv` from oneapi compiler, it solves the problem with crash on "-O0" opt level

**Review by diptorupd — APPROVED** [review]

File: numba_dppy/codegen.py [hunk 1]
```diff
@@ -38,9 +39,14 @@ def _optimize_functions(self, ll_module):
...
 
     def _optimize_final_module(self):
+        opt_level_option = config.OPT
+
         # Run some lightweight optimization to simplify the module.
         pmb = ll.PassManagerBuilder()
-        pmb.opt_level = 1
+
+        # Turn off any optimizations if OPT level is set to 0
+        pmb.opt_level = 0 if opt_level_option == 0 else 1
```
Comment (PokhodenkoSA) [inline numba_dppy/codegen.py:48]: Why `pmb.opt_level` is not equal to `config.OPT`?

**Comment by reazulhoque** [comment]
@PokhodenkoSA @diptorupd should we merge this?

**Comment by akharche** [comment]
@reazulhoque I think, it should be merge after problem with -O0 (addr space casting) is fixed. This code will fail if user changes option NUMBA_OPT to 0

**Comment by PokhodenkoSA** [comment]
> This code will fail if user changes option NUMBA_OPT to 0

It is up to user to use any NUMBA_OPT value. This PR only provides an option to configure.
If default NUMBA_OPT value is appropriate we could merge this PR.

**Review by PokhodenkoSA — APPROVED** [review]

**Pull request merged**